components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java

```diff
@@ -397,6 +427,10 @@ protected void initiateAuthenticationRequest(HttpServletRequest request, HttpSer
 
                 String queryString = getQueryString(authenticatorProperties);
                 if (StringUtils.isNotBlank(scopes)) {
+                    if (LoggerUtils.isDiagnosticLogsEnabled() && diagnosticLogBuilder != null) {
```

Comment:
`diagnosticLogBuilder != null` isn't this redundant?

Reply:
This check is added to remove the compiler warning